pkgs/dartpad_ui/lib/editor/editor.dart

```diff
@@ -177,10 +177,17 @@ class _EditorWidgetState extends State<EditorWidget> implements EditorService {
   @override
   void initState() {
     super.initState();
-
+    _autosaveTimer = Timer.periodic(const Duration(seconds: 5), _autosave);
     widget.appModel.appReady.addListener(_updateEditableStatus);
   }
 
+  Timer? _autosaveTimer;
+  void _autosave([Timer? timer]) {
+    final content = codeMirror?.getDoc().getValue();
+    if (content == null) return;
```

Comment:
I think it'd be more consistent to rely on the app model here and help if/when we move away from code mirror.

```suggestion
    final content = widget.appModel.sourceCodeController.text;
```

Reply:
Fixed, and I added a check for `isEmpty` instead of null